### ISSUE TITLE
addAfterChild adds before 'afterChild' instead of after

### ISSUE THREAD

**Issue by GoogleCodeExporter** [bctpl/tree-view-list-android#19]
```
addAfterChild(final T parent, final T newChild, final T afterChild),
when afterChild is not null, adds the new child before afterChald and not after.

As I understand it (I only looked at the code for this method so I don't know 
if there is other implications) the line:

final InMemoryTreeNode<T> added = node.add(
     index == -1 ? node.getChildrenListSize() : index, newChild, 
     visibility);

should be:

final InMemoryTreeNode<T> added = node.add(
     index == -1 ? node.getChildrenListSize() : index+1, newChild,
     visibility);

So the index of the added node will be 1 after 'afterChild' so it will appear 
after.

```

Original issue reported on code.google.com by `nicola.l...@googlemail.com` on 11 Jan 2012 at 12:40

**Comment by GoogleCodeExporter** [comment]
```
This issue was closed by revision 51409b85214d.
```

Original comment by `jarek.po...@polidea.pl` on 11 Jan 2012 at 8:10
- Changed state: **Fixed**

**Comment by GoogleCodeExporter** [comment]
```
Thanks. Good Catch!
```

Original comment by `jarek.po...@polidea.pl` on 11 Jan 2012 at 8:11
